cmd/cmd.go

```diff
@@ -803,15 +804,50 @@ func generate(cmd *cobra.Command, opts runOptions) error {
 	return nil
 }
 
-func RunServer(cmd *cobra.Command, _ []string) error {
-	host, port, err := net.SplitHostPort(strings.Trim(os.Getenv("OLLAMA_HOST"), "\"'"))
+func getHostIPAndPort() (string, string, error) {
+	hostStr := strings.Trim(strings.TrimSpace(os.Getenv("OLLAMA_HOST")), "\"'")
+	if hostStr == "" {
+		hostStr = "127.0.0.1:11434"
+	}
+
+	host, port, err := net.SplitHostPort(hostStr)
 	if err != nil {
-		host, port = "127.0.0.1", "11434"
-		if ip := net.ParseIP(strings.Trim(os.Getenv("OLLAMA_HOST"), "[]")); ip != nil {
-			host = ip.String()
+		var netErr *net.AddrError
+		if errors.As(err, &netErr) {
+			if netErr.Err == "missing port in address" {
+				host, port, err = net.SplitHostPort(hostStr + ":11434")
+				if err != nil {
+					return "", "", err
+				}
+			} else {
+				return "", "", err
+			}
+		} else {
+			return "", "", err
 		}
 	}
 
+	if host == "" {
+		host = "0.0.0.0"
+	}
+
+	if ip := net.ParseIP(host); ip == nil {
+		return "", "", fmt.Errorf("invalid IP address specified in OLLAMA_HOST")
+	}
```

Comment:
I've changed this to allow hostnames, although we might want to consider warning if someone uses a hostname since it will only bind to a single IP. From the `net.Listen()` docs:

> The address can use a host name, but this is not recommended, because it will create a listener for at most one of the host's IP addresses.